benchmark/single-source/Flatten.swift

```diff
@@ -0,0 +1,962 @@
+    let cl = f.reduce(into: CountLast(), CountLast.collect)
+    CheckResults((cl.count, cl.last!) == (flatSize, lastElement))
+  }
+}
+
+@inline(never)
+public func run_FlattenAnySeqSeqArray4_flatMap(_ N: Int) {
+  for _ in 1...N {
+    let input = AnySequence(sequenceArray4)
+
+    let f = input.flatMap { $0 }
+
+    let cl = f.reduce(into: CountLast(), CountLast.collect)
+    CheckResults((cl.count, cl.last!) == (flatSize, lastElement))
+  }
+}
+
+@inline(never)
+public func run_FlattenAnySeqSeqArray4_joined(_ N: Int) {
+  for _ in 1...N {
+    let input = AnySequence(sequenceArray4)
+
+    let f = input.joined()
+
+    let cl = f.reduce(into: CountLast(), CountLast.collect)
+    CheckResults((cl.count, cl.last!) == (flatSize, lastElement))
+  }
+}
+
+@inline(never)
+public func run_FlattenAnySeqSeqArray4_forinElem(_ N: Int) {
+  for _ in 1...N {
+    let input = AnySequence(sequenceArray4)
+
+    var f: [Int] = []
+
+    for array in input {
+      for element in array {
+        f.append(element)
+      }
+    }
+
+    let cl = f.reduce(into: CountLast(), CountLast.collect)
+    CheckResults((cl.count, cl.last!) == (flatSize, lastElement))
+  }
+}
+
+@inline(never)
+public func run_FlattenAnySeqSeqArray4_forinBatch(_ N: Int) {
+  for _ in 1...N {
+    let input = AnySequence(sequenceArray4)
+
+    var f: [Int] = []
+
+    for array in input {
+      f += array
+    }
+
+    let cl = f.reduce(into: CountLast(), CountLast.collect)
+    CheckResults((cl.count, cl.last!) == (flatSize, lastElement))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorVal_flatMap(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    let f = input.flatMap { [$0.a, $0.r, $0.g, $0.b] }
+
+    CheckResults((f.count, f.last!) == (flatSize, lastBlue))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorVal_map_joined(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    let f = input.map({ [$0.a, $0.r, $0.g, $0.b] }).joined()
+
+    CheckResults((f.count, f.last!) == (flatSize, lastBlue))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorVal_lazy_map_joined(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    let f = input.lazy.map({ [$0.a, $0.r, $0.g, $0.b] }).joined()
+
+    let cl = f.reduce(into: CountLast(), CountLast.collect)
+    CheckResults((cl.count, cl.last!) == (flatSize, lastBlue))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorVal_forinNaive(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    var f: [UInt8] = []
+
+    for c in input {
+      f.append(c.a)
+      f.append(c.r)
+      f.append(c.g)
+      f.append(c.b)
+    }
+
+    CheckResults((f.count, f.last!) == (flatSize, lastBlue))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorValUnsafe(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    let f = input.withUnsafeBufferPointer {
+      $0.baseAddress!.withMemoryRebound(to: UInt32.self, capacity: input.count) {
```

Comment:
Why are you viewing this at `UInt32` and doing bitwise arithmetic instead of viewing it as `UInt8`?

Reply:
As is described in the `.gyb` file, the pretend scenario for `Color` group is conversion from RGBA to ARBG pixel format. To do the rotation, I have to step in 32bit chunks. Or is there other way?

Reply:
I get that an RGBA->ARGB conversion is happening, although I would never guess from the function name and I don't see any comments.

I would have just used `withUnsafeBytes` instead of `withMemoryRebound` and looped over the indices `for i on stride(from: 0, to: N, by: 4)`.

Reply:
If there's some reason that you decided to write it one way versus another, maybe because you're trying to emulate the non-unsafe loop, then please comment that!

Reply:
The idea was to systematically extend the coverage beyond the `Int` tuple @gottesmm used. So I’ve added array of arrays, then I though I need to also cover struct and a class. To make it somewhat more realistically motivated, I thought of swizzling the byte order of pixel data. That also seems like a more realistically motivated scenario for the aspirational Unsafe version than the contrived `Tuple4.Unsafe` variant.

Can you please write the swizzling using `stride` for me? I don’t understand how to apply what you suggested.